packages/workshop-app/server/index.ts

```diff
@@ -1,146 +1,199 @@
 import fs from 'fs'
 import path from 'path'
+import { fileURLToPath, pathToFileURL } from 'url'
 import express from 'express'
+import chokidar from 'chokidar'
 import compression from 'compression'
 import morgan from 'morgan'
 import address from 'address'
 import closeWithGrace from 'close-with-grace'
-import ws from 'ws'
 import { createRequestHandler } from '@remix-run/express'
-import { getApps, getWorkshopRoot } from '../utils/apps.server'
-import { getWatcher } from '../utils/change-tracker'
-import { purgeRequireCache } from '../utils/purge-require-cache.server'
+import { type ServerBuild, broadcastDevReady } from '@remix-run/node'
+import { getApps, getWorkshopRoot } from '../utils/apps.server.ts'
+import getPort, { portNumbers } from 'get-port'
+import chalk from 'chalk'
 
-declare global {
-	var __server_close_with_grace_return__: ReturnType<typeof closeWithGrace>
-}
+/*
+// FIXME:
+	when restarting the server (from tsx watch):
+	  (some time the running port is unavailable when we get here again)
+		save current running port number
+		kill the running port, see kill function from remix.
+		https://github.com/remix-run/remix/blob/main/packages/remix-dev/devServer_unstable/index.ts
+		start the server again on the same port
 
-async function start() {
-	// get some caches warmed up
-	import('globby')
-	import('execa')
-	import('get-port')
-	import('p-map')
-	getApps()
+	do we need to do everithing again?
+		caches already warmed
+		all apps already cached
 
-	const { default: getPort, portNumbers } = await import('get-port')
-	const BUILD_DIR_FILE = path.join(process.cwd(), 'build/remix.js')
-	const workshopRoot = getWorkshopRoot()
+	do we need to close old watcher?
+*/
 
-	const app = express()
+// get some caches warmed up
+import('globby')
+import('execa')
+import('get-port')
+import('p-map')
 
-	app.use(compression())
+// FIXME: (maybe) - we can complie this file (like in epic-stack) to
```

Comment:
Yes, I definitely want this app to resemble epic stack where possible.

Reply:
I can move `build/server/index.js` to `server-build/index.js` then we can use `import * as remixBuild from "../build/index.js";`

but in kcdshop we use `import { getApps, getWorkshopRoot } from "../utils/apps.server.js";`

let me know what you prefer:
1. leave the code as it is now
2. move `build/server/index.js` to `server-build/index.js` and use dynamic import for `apps.server.js`
3. other ideas

Reply:
see https://github.com/epicweb-dev/kcdshop/pull/64/commits/ec3212d23e1b139092c6754eee0a1499f86f3212